pkg/build/strategies/sti/sti.go

```diff
@@ -452,11 +452,18 @@ func (b *STI) Execute(command string, user string, config *api.Config) error {
 	// and wait till all injections are uploaded into the container that runs the
 	// assemble script.
 	if len(config.Injections) > 0 && command == api.Assemble {
+		workdir, err := b.docker.GetImageWorkdir(config.BuilderImage)
+		if err != nil {
+			return err
+		}
+		if err := util.FixInjectionsWithRelativePath(workdir, &config.Injections); err != nil {
```

Comment:
@bparees just FYI, I was terribly wrong thinking that the WORKPATH is used automatically, it is not. The default for uploading files into container is set by Docker and it defaults to "/", which is not what we want. This should get the WORKDIR from the builder image and use it as a prefix for relative path...